common/src/main/java/org/keycloak/common/util/CertificateUtils.java

```diff
@@ -17,38 +17,24 @@
 
 package org.keycloak.common.util;
 
-import org.bouncycastle.asn1.x500.X500Name;
-import org.bouncycastle.asn1.x509.BasicConstraints;
-import org.bouncycastle.asn1.x509.ExtendedKeyUsage;
-import org.bouncycastle.asn1.x509.Extension;
-import org.bouncycastle.asn1.x509.KeyPurposeId;
-import org.bouncycastle.asn1.x509.KeyUsage;
-import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
-import org.bouncycastle.cert.X509CertificateHolder;
-import org.bouncycastle.cert.X509v1CertificateBuilder;
-import org.bouncycastle.cert.X509v3CertificateBuilder;
-import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
-import org.bouncycastle.cert.jcajce.JcaX509ExtensionUtils;
-import org.bouncycastle.operator.ContentSigner;
-import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
-
 import java.math.BigInteger;
 import java.security.KeyPair;
 import java.security.PrivateKey;
-import java.security.SecureRandom;
 import java.security.cert.X509Certificate;
-import java.util.Calendar;
-import java.util.Date;
+
+import org.keycloak.common.crypto.CryptoIntegration;
 
 /**
  * The Class CertificateUtils provides utility functions for generation of V1 and V3 {@link java.security.cert.X509Certificate}
  *
- * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
- * @author <a href="mailto:giriraj.sharma27@gmail.com">Giriraj Sharma</a>
- * @version $Revision: 2 $
  */
 public class CertificateUtils {
 
+    static {
```

Comment:
Is it possible to remove this initialization from static block? It seems to me that it is needed just because of the unit tests? So instead of calling static initialization here, it can be better if you just add this at the beginning of the unit test:
```
    @ClassRule
    public static CryptoInitRule cryptoInitRule = new CryptoInitRule();
```

See for example `DefaultSecureRandomTest` for the inspiration. I hope this will work to fix the tests.

Reply:
Added @Deprecated to static classes, I would prefer to open follow-up issue to remove static calls. Do you concur?

Reply:
Ok, I've created follow-up task for remove static initialization block https://github.com/keycloak/keycloak/issues/13128 . But I am not sure about the @deprecated annotation on static classes. IMO there is not any reason for these classes to be deprecated as they are still widely used from Keycloak codebase?

Just to doublecheck, I was referring only to remove static initialization block in the comment above, not to remove whole static class (or deprecate whole class).